exporters/metrics/src/test/java/com/google/cloud/opentelemetry/metric/EndToEndTest.java

```diff
@@ -0,0 +1,83 @@
+package com.google.cloud.opentelemetry.metric;
+
+import static com.google.cloud.opentelemetry.metric.FakeData.aGceResource;
+import static com.google.cloud.opentelemetry.metric.FakeData.aMonotonicLongDescriptor;
+import static com.google.cloud.opentelemetry.metric.FakeData.aProjectId;
+import static com.google.cloud.opentelemetry.metric.FakeData.anInstrumentationLibraryInfo;
+import static com.google.cloud.opentelemetry.metric.MetricTranslator.NANO_PER_SECOND;
+import static java.time.temporal.ChronoUnit.SECONDS;
+import static org.junit.Assert.assertEquals;
+
+import com.google.common.collect.ImmutableList;
+import io.opentelemetry.common.Labels;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.data.MetricData.LongPoint;
+import io.opentelemetry.sdk.metrics.export.MetricExporter.ResultCode;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.ServerSocket;
+import java.time.Instant;
+import java.util.ArrayList;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class EndToEndTest {
+
+  private static final String LOCALHOST = "127.0.0.1";
+
+  private Process mockServerProcess;
+  private MetricExporter exporter;
+  private MockCloudMetricClient mockClient;
+
+  @Before
+  public void setup() throws IOException {
```

Comment:
nit: Assuming that this setup duplicates the one from tracing test, shall we extract it to a base class or some static helper function to be used by both?

Reply:
Since the trace and metric tests are in different modules, I'll leave it for now as we discussed in chat :)